bin/varnishd/cache/cache.h

```diff
@@ -443,6 +443,9 @@ struct busyobj {
 	const char		*err_reason;
 
 	const char		*client_identity;
+
+	VTAILQ_ENTRY(busyobj)	cw_list;
+	pthread_cond_t		cw_cond;
```

Comment:
A `pthread_cond_t` is pretty heavy, can't we re-use one we already have, like `(struct boc).cond` or `(struct worker).cond`?

FTR, I have not checked the details and in particular I have not looked at the impact on code clarity, but we should we sure the footprint is worth it.

Reply:
OR, even better, couldn't the two `cw_` members live on the stack, because the wait only exists within `vbe_dir_getfd()` ?

Reply:
The original implementation used, I believe, `(struct boc).cond`.  But there were concerns raised about hijacking that `cond` for something like this. To alleviate those concerns it was given its own `cond`, which also helps code clarity (IMO).

Reply:
> couldn't the two `cw_` members be packed into a struct which lives on the stack

I'm not sure I understand.  You want to put a list shared across threads on the stack?

Reply:
Basically yes.
In terms of [queue(3)](https://man.netbsd.org/queue.3) which we use prefixed with `V`, there is a list head and entries linked with (double) pointers. As long as we make sure that we remove it from the list before the stack frame becomes invalid, we can sure put an entry on the stack.

Reply:
I think it is better where it is.

Reply:
If we end up moving it to the stack, I'd prefer seeing it in a followup commit to keep this port aligned with the original change.

No opinion otherwise.

Reply:
@drodden "I think it is better where it is" is not quite a convincing argument. My arguments are:

* The additional memory requirement for _every_ `struct busyobj` is relevant

```
(gdb) p sizeof(struct busyobj)
$1 = 440
(gdb) p sizeof(pthread_cond_t)
$2 = 48
```

* The cv is only needed in _that_ particular function, which is only called when queuing, so adding overhead to the general case (memory, initialization, destruction) is not justified if there is a sensible alternative option.

Re Dridi: sure, why not

Reply:
@nigoroll, no it was not a very good response and I know better than to post anything before fully waking up and having my morning tea.

If you'd like to move it, I'm with Dridi in doing as a follow up.

Reply:
Moved the `cw_*` members to the stack